### ISSUE TITLE
Restrict transitive dependencies when using eclipseMavenCentral

### ISSUE THREAD

**Issue by nedtwigg** [diffplug/goomph#149]
This commit doesn't change the build at all: https://github.com/diffplug/goomph/commit/c10d1edc56a86465574c4e8e965d1cd39e9550cd

Nevertheless, the build fails at this point due to an updated transitive dependency:
https://app.circleci.com/pipelines/github/diffplug/goomph/51/workflows/227c2a50-c5b2-4166-a0b8-6c18dda0cbd8/jobs/85

This is caused because eclipse artifacts specify wide compatibility ranges, and as new artifacts become available, they are not resolved "within the release", but instead they are resolved across the entire maven catalog.

We should figure this out somehow or other: https://docs.gradle.org/current/userguide/resolution_rules.html

**Comment by jmini** [comment]
I think I faced a similar issue. For each Eclipse release I link:https://jmini.github.io/ecentral/[created] Maven BOM files that can be used as [`platform` dependency](https://docs.gradle.org/current/userguide/java_platform_plugin.html) in Gradle.

With the corresponding repository added:
```
repositories {
    maven {
        url 'https://raw.githubusercontent.com/jmini/ecentral/1.0.0/repo'
    }
    mavenCentral()
}
```

You can do:
```
eclipseMavenCentral {
    release '4.14.0', {
        implementation 'org.eclipse.ui'
        implementation 'org.eclipse.core.runtime'

        // specify this to add the native jars for this platform
        useNativesForRunningPlatform()
    }
}

dependencies {
    api platform('fr.jmini.ecentral:eclipse-platform-dependencies:4.14')

    implementation project(':fr.jmini.eclipse.core')
}
```

And you will get only the jars corresponding to the `2019-12` Eclipse release.

**Comment by nedtwigg** [comment]
This is resolved in Goomph `3.31.0` with `constrainTransitivesToThisRelease()`.  e.g.

```gradle
eclipseMavenCentral {
    release '4.14.0', {
        implementation 'org.eclipse.ui'
        implementation 'org.eclipse.core.runtime'
        useNativesForRunningPlatform()
        constrainTransitivesToThisRelease()
    }
}
```

**Comment by nedtwigg** [comment]
ec3e792e04efe99d878fce90f5acfb9947283e9b
